File: kotlin/app/src/main/kotlin/com/example/repositories/JokeRepository.kt
package com.example.repositories

import com.example.models.Joke
import com.example.models.JokeCreate

interface JokeRepository {
    suspend fun getAll(): List<Joke>
    suspend fun getById(id: Int): Joke?
    suspend fun create(create: JokeCreate): Joke
    suspend fun update(id: Int, joke: Joke): Boolean
    suspend fun delete(id: Int): Boolean
    suspend fun getRandom(category: String? = null): Joke?
    suspend fun getBySource(source: String): List<Joke>
    suspend fun incrementLol(id: Int): Int
    suspend fun incrementGroan(id: Int): Int
}

File: swift/Sources/App/Domain/JokeRepository.swift
import Foundation

protocol JokeRepository {
    func create(joke: JokeCreate) async throws -> Joke
    func getById(_ id: String) async throws -> Joke?
    func getAll(source: String?, category: String?, limit: Int, offset: Int) async throws -> [Joke]
    func update(id: String, joke: JokeCreate) async throws -> Joke?
    func delete(id: String) async throws -> Bool
    func getRandom(category: String?) async throws -> Joke?
    func bumpLol(id: String) async throws -> CountsResponse?
    func bumpGroan(id: String) async throws -> CountsResponse?
}
